src/xbase/layout/cap/mim.py

```diff
@@ -0,0 +1,149 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2019 Blue Cheetah Analog Design Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module defines various MOM cap related templates."""
+
+from typing import Any, Dict, Optional, Mapping, List, Tuple, Type
+import math 
+
+from pybag.core import BBox, BBoxArray
+from pybag.enum import RoundMode, Orient2D
+
+from bag.util.immutable import Param
+from bag.design.module import Module
+from bag.layout.routing.grid import FillConfigType
+from bag.layout.template import TemplateDB, TemplateBase
+
+from xbase.layout.fill.base import DeviceFill
+from xbase.schematic.momcap_core import xbase__momcap_core
+from xbase.layout.data import LayoutInfo, draw_layout_in_template
+from xbase.layout.cap.tech import MIMTech, MIMLayInfo
+
+
+class MIMCapCore(TemplateBase):
+    """MIMCap core
+    """
+
+    def __init__(self, temp_db: TemplateDB,
+                 params: Param, **kwargs: Any) -> None:
+        TemplateBase.__init__(self, temp_db, params, **kwargs)
+
+    @classmethod
+    def get_schematic_class(cls) -> Optional[Type[Module]]:
+        return xbase__momcap_core
+
+    @classmethod
+    def get_params_info(cls) -> Dict[str, str]:
```

Comment:
This method is completely wrong. This is supposed to return Dict[str, str], and the contents are string explanations of the different parameters. The default values of the parameters are specified in get_default_param_values().